src/WebJobs.Script/ExtensionBundle/ExtensionBundleHttpExtensions.cs

```diff
@@ -0,0 +1,73 @@
+// Centralized helper for reading and sanitizing the x-azure-ref header for Extension Bundle downloads.
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Script.ExtensionBundle
+{
+    internal static class ExtensionBundleHttpExtensions
+    {
+        internal const string AzureRefHeaderName = "x-azure-ref";
+        private const int MaxAzureRefLength = 128;
+
+        internal static string GetAzureRef(this HttpResponseMessage response)
```

Comment:
Please use the `TryGet***` pattern for both of these methods. Avoid returning `null` strings.

Reply:
Done